src/dns.rs

```diff
@@ -176,8 +187,11 @@ mod tests {
     fn parse_str() {
         let mut dns = Dns::new();
         dns.names.insert("foo".into(), "127.0.0.1".parse().unwrap());
```

Comment:
Can we change this to not be `127.0.0.1` and use lookup to generate an ip per your changes?